lib/mix/phoenix/schema.ex

```diff
@@ -254,11 +256,29 @@ defmodule Mix.Phoenix.Schema do
         :time           -> ~T[14:00:00]
         :time_usec      -> ~T[14:00:00.000000]
         :uuid           -> "7488a646-e31f-11e4-aace-600308960662"
-        :utc_datetime   -> DateTime.utc_now() |> DateTime.add(-100_000) |> DateTime.truncate(:second)
-        :utc_datetime_usec -> DateTime.add(DateTime.utc_now(), -100_000)
-        :naive_datetime -> NaiveDateTime.utc_now() |> NaiveDateTime.add(-100_000) |> NaiveDateTime.truncate(:second)
-        :naive_datetime_usec -> NaiveDateTime.add(NaiveDateTime.utc_now, -100_000)
-        _               -> "some #{key}"
+        :utc_datetime   ->
+          DateTime
+          |> build_datetime()
```

Comment:
@josevalim I feel this code is getting too complicated but I couldn't think in another way to build the date here and assert them in the tests. The issue is that we need to force the seconds and microseconds in order to proper assert `datetime` and `naive_datetime`.

Another option would be removing the date from the assertion, but I am afraid we are going to lose test coverage with that. Do you have something better in mind?

Reply:
Looks like the seconds is a constant:
```
seconds = -24 * 3600
```

Then it means, it doesn't need to be here. 

Then I think it's possible to create something like:

```elixir
# ...
:utc_date_time -> build_utc_date_time()
# ...

def build_utc_date_time(), 
  do: DateTime.utc_now() |> DateTime.add(@seconds) |> DateTime.truncate(:second)

# apply the same pattern for the other data constructors
```

The naming here could be improved, for example, I have no idea what the `seconds` represent here.

I hope it helps 🙏  
Great job

Reply:
Humm... I updated the code inspired by your version. It sounds better now. Wdyt?

One thing that I don't like is that I will need to do the same in the tests, or I could expose those functions and use there 🤔.